src/parser/parser.py

```diff
@@ -0,0 +1,64 @@
+"""Parses a PDF document.
+"""
+import logging
+
+
+import slate
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Article:
+    """Class for storing articles.
+    """
+
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+def count_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles 
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
+    while i < len(pdf_text) - 1:
+        if (pdf_text[i] == "artículo" or pdf_text[i] == "articulo") and (
+                pdf_text[i + 1] == str(article_count) + ".-"
+                or pdf_text[i + 1] == str(article_count) + "-"
+                or pdf_text[i + 1] == str(article_count) + "."
+        ):
+            logging.info("Article #" + str(article_count) + " recognized!")
+            articles.append(Article(article_count, article_text))
+            article_text = ""
+            article_count += 1
+            i += 1
+        else:
+            article_text += " " + pdf_text[i]
+        i += 1
+    for article in articles:
+        logging.debug("Article: " + str(article.number - 1) + " Text: " + article.text)
+    return articles
+
+
+# Acts as main
```

Comment:
I'm guessing all of this pattern matching is aimed specifically at the document we're parsing right? Like looking for either **artículo** or **articulo**.

I'm pretty sure that each document is gonna have its own patterns and is gonna be presented different, so I guess this makes sense. Although this is definitely out of the scope of what we're aiming for, I think it'd be good to do some brainstorming to think of ways to come up with a general algorithm that does a first clean up.

Imagine having to rewrite this every time we want to add a new geographical zone. 🥵

Reply:
Yes, it is specific to our document. I understand that it would be nice to parse many documents with ease, but I see that as post MVP. Right now I see our focus to be to parse this document accurately.